client/post-editor/editor-sidebar/feedback-view.jsx

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes, PureComponent } from 'react';
+import { localize } from 'i18n-calypso';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import FeedbackSidebarHeader from './feedback-header';
+import FeedbackRequestForm from './feedback-request-form';
+import FeedbackList from './feedback-list';
+import SidebarFooter from 'layout/sidebar/footer';
+
+// TODO: Find a clearer word than "share" for the owner of feedback and use it in subcomponent and CSS classes
+class FeedbackView extends PureComponent {
+	static propTypes = {
+		translate: PropTypes.func,
+		close: PropTypes.func,
+		sharedLinks: PropTypes.array
+	}
+
+	constructor() {
+		super();
+		this.state = { openFeedbackCount: 0 };
+	}
+
+	onToggleFeedback = isOpen => {
+		const { openFeedbackCount } = this.state;
+
+		this.setState( {
+			openFeedbackCount: openFeedbackCount + ( isOpen ? 1 : -1 )
+		} );
+	};
+
+	haveShares() {
+		return this.props.sharedLinks.length > 0;
+	}
+
+	isAllFeedbackClosed() {
+		return this.state.openFeedbackCount === 0;
+	}
```

Comment:
I agree with you about extracting functions and then also about the functions not really making things clearer. I'll drop the helpers.

Also, it's cool to see your example with `( booleanExpression && ( reactElement ) )`. I wasn't sure whether React would render "false" for a `false` value, but my testing says that it doesn't, which is cool. I did discover that `( zero && ( reactElement ) )` results in a zero being rendered, so the expressions need to evaluate to real boolean values, not just falsey values.